tests/test_blueprints.py

```diff
@@ -308,6 +308,34 @@ def handler2(request):
     assert "test_bp_with_host_list.test_bp_host.handler2" in route_names
 
 
+def test_bp_with_auto_name_generate(app: Sanic):
+    bp = Blueprint(
+        "test_bp_host",
+        url_prefix="/test1",
+        host=["example.com", "sub.example.com"],
+        generate_name=True,
+    )
+
+    @bp.route("/")
+    def handler1(request):
+        return text("Hello")
+
+    @bp.route("/", host=["sub1.example.com"])
+    def handler2(request):
+        return text("Hello subdomain!")
+
+    @bp.route("/route_with_name", methods=["GET", "POST"], name="handler3")
+    def handler3(request):
+        return text("Hello subdomain!")
+
+    app.blueprint(bp)
+
+    route_names = [r.name for r in app.router.routes]
+    assert "test_bp_with_auto_name_generate.test_bp_host.handler1_GET_/" in route_names
+    assert "test_bp_with_auto_name_generate.test_bp_host.handler2_GET_/" in route_names
```

Comment:
We do not want slashes in route names.

Reply:
Its replaced with 'S'
also, can you please merge this one? on Python 3.12, tests will fail in other case: https://github.com/sanic-org/sanic-testing/pull/74